pkg/controller/metrics/metrics_controller.go

```diff
@@ -55,12 +55,17 @@ func NewController() Controller {
 
 	return &controller{
 		Client: client.NewClient(),
+		API:&API{
+			Client:  http.DefaultClient,
+			baseURL: "http://localhost:9090",
+		},
 	}
 }
 
 type controller struct {
 	client.Client
 	DockInfo *model.DockSpec
+	API *API
```

Comment:
Please elaborate why this API struct is added here.

Reply:
controller has to connect prometheus client for some GET/POST operations. API strcut is to represent the client. This will also help us to mock the client.